src/instructlab/model/list.py

```diff
@@ -0,0 +1,151 @@
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    is_flag=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    data = []
+    for directory in model_dirs:
+        for entry in Path(directory).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                model, age, size = _analyze_gguf(entry)
+                data.append([model, age, size])
+            elif entry.is_dir():
+                for m in _analyze_dir(entry, list_checkpoints, directory):
+                    data.append(m)
+    print_table(["Model Name", "Last Modified", "Size"], data)
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MiB, GiB)
+def convert_bytes_to_proper_mag(f_size: float):
+    magnitudes = ["KiB", "MiB", "GiB", "TiB"]
+    magnitude = "B"
+    for mag in magnitudes:
+        if f_size >= 1024:
+            magnitude = mag
+            f_size /= 1024
+        else:
+            return f_size, magnitude
+    return f_size, magnitude
+
+
+# print_table displays the models on the system in a nicely formatted table
+def print_table(headers: list[str], data: list[list[str]]):
+    column_widths = [
+        max(len(str(row[i])) for row in data + [headers]) for i in range(len(headers))
+    ]
+    # Print separator line between headers and data
+    horizontal_lines = ["-" * (width + 2) for width in column_widths]
+    joining_line = "+" + "+".join(horizontal_lines) + "+"
+    print(joining_line)
+    outputs = []
+    for header, width in zip(headers, column_widths, strict=False):
+        outputs.append(f" {header:{width}} ")
+    print("|" + "|".join(outputs) + "|")
+    print(joining_line)
+    for row in data:
+        outputs = []
+        for item, width in zip(row, column_widths, strict=False):
+            outputs.append(f" {item:{width}} ")
+        print("|" + "|".join(outputs) + "|")
+    print(joining_line)
+
+
+AnalyzeResultGGUF = list[str]
+AnalyzeResultDir = list[list[str]]
+
+
+def _analyze_gguf(entry: Path) -> AnalyzeResultGGUF:
+    # stat the gguf, add it to the table
+    stat = Path(entry.absolute()).stat(follow_symlinks=False)
+    f_size = stat.st_size
+    f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+    # add to table
+    modification_time = os.path.getmtime(entry.absolute())
+    modification_time_readable = time.strftime(
+        "%Y-%m-%d %H:%M:%S", time.localtime(modification_time)
+    )
+    return [entry.name, modification_time_readable, f"{f_size:.1f} {magnitude}"]
+
+
+def _analyze_dir(
```

Comment:
do we follow symlinks?

Reply:
nope `stat = Path(full_file).stat(follow_symlinks=False)`

I think by default `os.walk` doesn't either?

Reply:
Aren't we using symlinks from cache to model dir? @RobotSail do you remember? Thanks

Reply:
OCI downloaded models are also symlinked from cache to model dir 
do we need to follow them to get metadata about the file, or is that stuff carried over in the symlink?

Reply:
cc @mbestavros

Reply:
speaking to oleg and mark now, they will follow up here

Reply:
@RobotSail and I hacked a bit on model paths last week - IIRC the model loaders ran into issues when we provided symlinks to them. I'd verify that but that would be the strongest reason I think

Reply:
I think we should get this in as is, and follow up with enhancements IMO